src/main/docs/guide/httpServer/byteBody.adoc

```diff
@@ -0,0 +1,134 @@
+When an api:io.micronaut.http.HttpRequest[] implements api:io.micronaut.http.ServerHttpRequest[], the new `byteBody()`
+method returns a api:io.micronaut.http.body.ByteBody[] with buffered, reactive and blocking APIs.
+
+When an HTTP request comes in, it starts out with an unparsed and unclaimed stream of bytes as its `byteBody()`. After
+all request filters have run, typically an argument binder matching the `@Body` parameter of the controller will
+"claim" the `byteBody()` and e.g. parse the JSON. Finally, the body is closed at the end of the request lifecycle,
+discarding any data if it has not been claimed by the argument binder.
+
+=== Primary operations
+
+`ByteBody` itself does not offer direct access to the data. To begin processing, there must be a _primary
+operation_ that converts the body into another form that can be used in the application programming model.
+
+A normal `ByteBody` has two groups of streaming primary operations. `toInputStream()` gives access to the body
+as a regular `InputStream`. The `toByteArrayPublisher()` and `toByteBufferPublisher()` methods return a reactive stream
+of byte arrays or `ByteBuffer`s.
+
+WARNING: `InputStream` is blocking API, and the netty event loop must never be blocked. If you wish to read from the
+body using an `InputStream`, take care to do so only on another thread, or to annotate your filter with
+`@ExecuteOn(TaskExecutors.BLOCKING)`.
+
+If you need full access to the body, the `buffer()` method returns a `CompletableFuture` that completes with an
+`AvailableByteBody` when the full body has been received. `AvailableByteBody` has a few more convenient
+primary operations: `toByteArray()`, `toByteBuffer()` and `toString(Charset)`.
+
+Buffering is limited to the number of bytes configured in the `micronaut.server.max-request-buffer-size` property.
+
+=== Splitting
+
+Because the framework will not buffer the whole body in memory by default, after an ByteBody has been claimed (a
+primary operation has been performed), the data is "gone", and the same ByteBody cannot be claimed again. That
+means that if a filter were to claim the `ServerHttpRequest.byteBody()` directly (e.g. to print it to a log),
+controllers could not access it anymore. The argument binder for the `@Body` argument would throw an exception.
+
+To resolve this exclusivity problem, an `ByteBody` can be _split_ before it is claimed. The split operation
+essentially duplicates the body stream so that the two consumers (logging and argument binding) can process it
+independently. A body can be `split` any number of times, but only before the primary operation.
+
+While `ServerHttpRequest.byteBody()` returns a normal `ByteBody` -- cleanup is done by the HTTP server if the
+body is not consumed--the body returned by `split` is a `CloseableByteBody`. The caller *must* ensure that the
+new instance is closed, otherwise there can be resource and memory leaks, stalled connections, or other issues.
+
+==== Backpressure
```

Comment:
This should be in a different file to simplify linking and navigation to the docs.